src/bundle.c

```diff
@@ -1625,6 +1625,136 @@ gboolean check_bundle_payload(RaucBundle *bundle, GError **error)
 	return res;
 }
 
+gboolean replace_signature(RaucBundle *bundle, const gchar *insig, const gchar *outpath, gboolean verify, GError **error)
+{
+	g_autoptr(GFile) outfile = NULL;
+	g_autoptr(RaucBundle) ibundle = NULL;
+	g_autoptr(GFile) ifile = NULL;
+	GFileOutputStream* istream = NULL;
+	g_autofree gchar *iname = NULL;
+	g_autofree gchar *tmpdir = NULL;
+	g_autoptr(GBytes) sig = NULL;
+	GError *ierror = NULL;
+	gboolean res = FALSE;
+	gssize size;
+
+	g_return_val_if_fail(bundle != NULL, FALSE);
+	g_return_val_if_fail(outpath != NULL, FALSE);
+	g_return_val_if_fail(insig != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	r_context_begin_step("replace_signature", "Replacing bundle signature", 1);
+
+	tmpdir = g_dir_make_tmp("rauc-replace-XXXXXX", &ierror);
+	if (tmpdir == NULL) {
+		g_propagate_prefixed_error(error, ierror,
+				"Failed to create tmp dir: ");
+		res = FALSE;
+		goto out;
+	}
+
+	iname = g_build_filename(tmpdir, "_bundle.raucb", NULL);
+	ifile = g_file_new_for_path(iname);
```

Comment:
I'd use a temporary file in the same directory (perhaps by appending `.tmp`), as this allows an atomic move by using g_rename(). Similar to `mksquashfs()` you should check if the destination file already exists.

Reply:
Ok but if destination already exists, should I raise an error or should I overwrite the destination to have the same behavior as `resign`

Reply:
It should raise an error. Overwriting in `rauc resign` sounds like a bug. :)